specs/patterns.md

```diff
@@ -777,3 +777,64 @@ Data packages can also be declared inline in the data catalog:
 ### Implementations
 
 None known.
+
+## Table Schema: Unique constraint
+
+### Overview
+
+A `primaryKey` uniquely identifies each row in a table. Per SQL standards, it
+cannot contain `null` values. This pattern implements the SQL UNIQUE constraint
+by introducing a `uniqueKeys` array, defining one or more row uniqueness
+constraints which do support `null` values.
+
+### Specification
+
+The `uniqueKeys` property, if present, `MUST` be an array. Each entry in the
+array must be a `uniqueKey`. A `uniqueKey` `MUST` be an object and `MUST` have
+the following properties:
+
+- `fields` – A string or array (structured as per `primaryKey`) specifying the
+  resource field or fields required to be unique for each row in the table.
+- `nullUnique` – A boolean that dictates whether `null` are considered unique
```

Comment:
As per discussion i think we could leave this option out and state that they are considered unique by default. For now we could merge with this and update.